src/T2DMap.cpp

```diff
@@ -613,6 +613,65 @@ void T2DMap::addSymbolToPixmapCache(const QString key, const bool gridMode)
     }
 }
 
+/*
+ * Helper used to size fonts establishes the font size to use to draw the given
+ * sample text centralized inside the given boundary (or rather the boundary
+ * reduced by the margin (0-40) as a percentage). This margin is defaulted to
+ * 20%.
+ */
+bool T2DMap::sizeFontToFitTextInRect( QFont & font, const QRectF & boundaryRect, const QString & text, const quint8 percentageMargin )
+{
+    QFont _font = font;
+
+    if (percentageMargin > 80) {
+        qWarning() << "T2DMap::sizeFontToFitTextInRect(...) percentage margin" << percentageMargin << "exceeded maximum (40%) !";
+        return false;
+    }
+    if (text.isEmpty()) {
+        qWarning() << "T2DMap::sizeFontToFitTextInRect(...) called with no sample text!";
```

Comment:
Why does this need to be a warning?

I'm just mindful of stdout being filled with a ton of meaningless text we need to train ourselves to ignore.

Reply:
It indicates a coding error to the developer - they shouldn't be trying to size a box to fit some text without *any* text..!

Reply:
So it should be an assert then?

Reply:
No need to terminate (with extreme prejudice) the entire application - it just aborts making any change to the supplied reference `QFont` argument...